### DISCUSSION TITLE
dark mode theme

### DISCUSSION THREAD

**Discussion by brunoh-android** [mapsforge/mapsforge#1386] [Q&A] [answered]
How to implement dark theme in maps? Preferably on mobile kotlin

I already searched but didn't find anything about it

**Comment by devemux86** [comment, marked as answer]
- You can create a theme with different colors ([render theme documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/Rendertheme.md))
- Use one of the available color filters:
```java
mapView.getModel().displayModel.setFilter(filter);
```
- Use a [ThemeCallback](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/ThemeCallback.java) to adjust the colors per `RenderInstruction`:
```java
mapView.getModel().displayModel.setThemeCallback(themeCallback);
```

**Reply by brunoh-android** [reply]
I didn't quite understand the idea, I'm still a beginner.

I tried to create a theme only with a different background color, passing only the background color, but on android it does not find this renderTheme.

Do I have to download some specific mapForge file?

**Reply by devemux86** [reply]
It depends where the theme is stored.

You can see the [getting started example](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java) for how to use Mapsforge
and the [mapsforge-samples-android](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-android) with many more examples.
(like the [MapsforgeMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/MapsforgeMapViewer.java))

**Reply by brunoh-android** [reply]
Got it, thanks for the help.

that way I arrived at the expected

if (isDarkMode)mapView.model.displayModel.filter = Filter.INVERT;




![Screenshot_20230119_100938](https://user-images.githubusercontent.com/67665152/213451188-4db47f55-0bc5-4500-83ed-c59bb14cbef0.png)
